starter-core/src/main/java/io/micronaut/starter/feature/microstream/MicroStream.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.microstream;
+
+import io.micronaut.core.annotation.NonNull;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class MicroStream implements Feature {
+
+    public static final String NAME = "microstream";
+    private static final String MICRONAUT_MICROSTREAM_GROUP_ID = "io.micronaut.microstream";
+
+    @Override
+    @NonNull
+    public String getName() {
+        return NAME;
+    }
+
+    @Override
+    public boolean isPreview() {
+        return true;
+    }
+
+    @Override
+    public String getTitle() {
+        return "MicroStream";
+    }
+
+    @Override
+    @NonNull
+    public String getDescription() {
+        return "Adds support for using MicroStream with Micronaut";
+    }
+
+    @Override
+    public boolean supports(ApplicationType applicationType) {
+        return true;
+    }
+
+    @Override
+    public String getCategory() {
+        return Category.DATABASE;
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.getConfiguration().put("microstream.storage.main.storage-directory", "/tmp/microstream");
```

Comment:
I don't we should add configuration entries. There is really not sensible defaults. Windows users may not have a `/tmp` folder. What do you think if we removed both lines: 

```
        generatorContext.getConfiguration().put("microstream.storage.main.storage-directory", "/tmp/microstream");
Write
        generatorContext.getConfiguration().put("microstream.storage.main.root-class", "java.util.ArrayList");
```